step-functions-data-science-sdk/automate_model_retraining_workflow/automate_model_retraining_workflow.ipynb

```diff
@@ -0,0 +1,983 @@
+    "4. Choose **Next** until you can enter a **Role name**.\n",
+    "5. Enter a name such as `query_training_status-role` and then select **Create role**.\n",
+    "\n",
+    "Next, attach policies to the role you created. The following steps attach policies that provides Lambda access to S3 and read-only access to SageMaker.\n",
+    "\n",
+    "1. Under the **Permissions** tab, click **Attach Policies**.\n",
+    "2. In the search box, type **SageMaker** and select **AmazonSageMakerReadOnly** from the populated list.\n",
+    "3. In the search box type **AWSLambda** and select **AWSLambdaBasicExecutionRole** from the populated list.\n",
+    "4. Choose **Attach policy**. You will then be redirected to the details page for the role.\n",
+    "5. Copy the **Role ARN** at the top of the **Summary**.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# paste the query_training_status-role role ARN from above\n",
+    "lambda_role = ''"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Prepare the Dataset\n",
+    "This notebook uses the XGBoost algorithm to automate the classification of unhappy customers for telecommunication service providers. The goal is to identify customers who may cancel their service soon so that you can entice them to stay. This is known as customer churn prediction.\n",
+    "\n",
+    "The dataset we use is publicly available and was mentioned in the book [Discovering Knowledge in Data](https://www.amazon.com/dp/0470908742/) by Daniel T. Larose. It is attributed by the author to the University of California Irvine Repository of Machine Learning Datasets."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "project_name = 'ml_deploy'\n",
+    "\n",
+    "data_source = sagemaker.s3.S3Uploader.upload(local_path='./data/customer-churn.csv',\n",
+    "                               desired_s3_uri='s3://{}/{}'.format(bucket, project_name),\n",
+    "                               session=session)\n",
+    "\n",
+    "train_prefix = 'train'\n",
+    "val_prefix = 'validation'\n",
+    "\n",
+    "train_data = 's3://{}/{}/{}/'.format(bucket, project_name, train_prefix)\n",
+    "validation_data = 's3://{}/{}/{}/'.format(bucket, project_name, val_prefix)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create Resources\n",
+    "In the following steps we'll create the Glue job and Lambda function that are called from the Step Functions workflow."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create the AWS Glue Job"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "glue_script_location = sagemaker.s3.S3Uploader.upload(local_path='./code/glue_etl.py',\n",
```

Comment:
[Minor] Consider importing this so it's cleaner to write:
```
from sagemaker.s3 import S3Uploader
data_source = S3Uploader.upload([...])
```

Reply:
Good idea. Resolved in the most recent commit.